### PR TITLE
Add a dynamic inventory script for Gandi

### PR DESCRIPTION
**Pull request by PabloPie**
##### SUMMARY

Add a dynamic inventory script for Gandi IaaS + Fix naming in Gandi cloud module

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME

 - contrib/inventory/gandi.py
- lib/ansible/modules/cloud/gandi/gandi_vlan.py
##### ANSIBLE VERSION

```
2.4.0
```

### CODE REVIEW

File: lib/ansible/modules/cloud/gandi/gandi_iface.py [hunk 1]
```diff
@@ -89,19 +89,19 @@
     sys.exit(1)
 
 
-# Load in the libcloud secrets file
-try:
-    import secrets
-except ImportError:
-    secrets = None
+# # Load in the libcloud secrets file
```
Comment (gdchamal) [inline lib/ansible/modules/cloud/gandi/gandi_iface.py]: remove the code, not comment it (versioning permit to find it back)

File: lib/ansible/modules/cloud/gandi/gandi_vps.py [hunk 2]
```diff
@@ -272,7 +268,7 @@ def stop_instances(module, driver, instance_names):
 
             except GandiException as e:
                 msg = 'Unexpected error when starting instance %s' % name
-                msg = msg + e
+                msg = msg + str(e)
```
Comment (gdchamal) [inline lib/ansible/modules/cloud/gandi/gandi_vps.py]: Why this str(e) ?

Side note, variable name should be at least 3 characters -> rename e to exc for better python style

File: lib/ansible/modules/cloud/gandi/gandi_vps.py [hunk 3]
```diff
@@ -377,7 +373,7 @@ def create_instances(module, driver, instance_names):
 
     lc_image = get_image(driver, image, lc_location)
     if not lc_image:
-        lc_image = get_volume(driver, image, lc_location)
+        lc_image = get_volume(driver, image)
```
Comment (gdchamal) [inline lib/ansible/modules/cloud/gandi/gandi_vps.py]: I think missing lc_location can break things internal to gandi.
The get_volume should accept an optional lc_location parameter, default it to None

**Review by gdchamal — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/gandi/gandi_vps.py (hunk 2 above)
Comment (PabloPie) [reply lib/ansible/modules/cloud/gandi/gandi_vps.py]: If I remember correctly I had trouble concatenating without an explicit cast to string.

Comment (gdchamal) [reply lib/ansible/modules/cloud/gandi/gandi_vps.py]: Replace %s by %r should be better (it's the representation of the object e, so a string)

File: lib/ansible/modules/cloud/gandi/gandi_vps.py (hunk 3 above)
Comment (PabloPie) [reply lib/ansible/modules/cloud/gandi/gandi_vps.py]: The call to the api in the libcloud module is never made with any optional parameters, should we also update it?

Comment (gdchamal) [reply lib/ansible/modules/cloud/gandi/gandi_vps.py]: yes we must propagate this optional parameter

**Review by gdchamal — APPROVED** [review]

**Pull request merged**